core/src/test/java/org/opensearch/sql/planner/physical/NestedOperatorTest.java

```diff
@@ -0,0 +1,328 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.planner.physical;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.contains;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.when;
+import static org.opensearch.sql.data.model.ExprValueUtils.collectionValue;
+import static org.opensearch.sql.data.model.ExprValueUtils.tupleValue;
+import static org.opensearch.sql.data.type.ExprCoreType.STRING;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.opensearch.sql.data.model.ExprNullValue;
+import org.opensearch.sql.data.model.ExprValue;
+import org.opensearch.sql.expression.ReferenceExpression;
+
+@ExtendWith(MockitoExtension.class)
+class NestedOperatorTest extends PhysicalPlanTestBase {
+  @Mock
+  private PhysicalPlan inputPlan;
+
+  private final ExprValue testData = tupleValue(
+      Map.of(
+          "message",
+          collectionValue(
+              List.of(
+                  Map.of("info", "a"),
+                  Map.of("info", "b"),
+                  Map.of("info", "c")
+              )
+          ),
+          "comment",
+          collectionValue(
+              List.of(
+                  Map.of("data", "1"),
+                  Map.of("data", "2"),
+                  Map.of("data", "3")
+              )
+          )
+      )
+  );
+
+
+  private final ExprValue testDataWithSamePath = tupleValue(
+      Map.of(
+          "message",
+          collectionValue(
+              List.of(
+                  Map.of("info", "a", "id", "1"),
+                  Map.of("info", "b", "id", "2"),
+                  Map.of("info", "c", "id", "3")
+              )
+          )
+      )
+  );
+
+  private final ExprValue nonNestedTestData = tupleValue(
+      Map.of(
+          "message", "val"
+      )
+  );
+
+  private final ExprValue missingArrayData = tupleValue(
+      Map.of(
+          "missing",
+          collectionValue(
+              List.of("value")
+          )
+      )
+  );
+
+  @Test
+  public void nested_one_nested_field() {
+    when(inputPlan.hasNext()).thenReturn(true, false);
+    when(inputPlan.next())
+        .thenReturn(testData);
+
+    Set<String> fields = Set.of("message.info");
```

Comment:
what if nested("message"), not valid?

Reply:
This will fail the semantic check in [NestedAnalyzer](https://github.com/opensearch-project/sql/blob/22c9bfb084d1aca9eedd93d8cb0448e7bb546935/core/src/main/java/org/opensearch/sql/analysis/NestedAnalyzer.java#L82-L101)